client/capstone-app/src/pages/MapPage.jsx

```diff
@@ -0,0 +1,138 @@
+    const {currentCoordinates, destinationCoordinates} = useParams()
+    const [error, setError] = useState(null)
+
+    function convertCoordinates(coordString) {
+      return coordString.split(',').map(Number);
+    }
+
+    const fetchDirections = async () => {
+        try {
+        const response = await fetch(`https://api.openrouteservice.org/v2/directions/driving-car?api_key=5b3ce3597851110001cf6248d2ab3de842a1474a865cff3fff2e2787&start=${currentCoordinates}&end=${destinationCoordinates}`, {
+          method: 'GET',
+          headers: {
+            'Accept': 'application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8'
+          },
+        });
+        const data = await response.json();
+        setDirections(data)
+      } catch (error) {
+        console.error(error);
+      }
+    }
+
+    const currentCoordArray = convertCoordinates(currentCoordinates)
+    const destCoordArray = convertCoordinates(destinationCoordinates)
+
+  const fetchCurrentRadius = async () => {
+    try {
+      const response = await fetch('https://api.openrouteservice.org/v2/isochrones/foot-walking', {
+      method: 'POST',
+      headers: {
+        'Accept': 'application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8',
+        'Authorization': '5b3ce3597851110001cf6248d2ab3de842a1474a865cff3fff2e2787',
+        'Content-Type': 'application/json; charset=utf-8'
+      },
+      body: JSON.stringify({
+        locations: [currentCoordArray],
+        range: [0, 1],
+        location_type: 'start',
+        range_type: 'distance',
+        units: 'mi'
+      })
+    })
+      console.log('Status:', response.status);
+      const data =  await response.json();
+      setCurrentRadius(data)
+    }
+    catch(error) {
+      console.error('Error:', error);
+    };
+  }
+
+    const fetchDestinationRadius = async () => {
+      try {
+        const response = await fetch('https://api.openrouteservice.org/v2/isochrones/foot-walking', {
+        method: 'POST',
+        headers: {
+          'Accept': 'application/json, application/geo+json, application/gpx+xml, img/png; charset=utf-8',
+          'Authorization': '5b3ce3597851110001cf6248d2ab3de842a1474a865cff3fff2e2787',
+          'Content-Type': 'application/json; charset=utf-8'
+        },
+        body: JSON.stringify({
+          locations: [destCoordArray],
+          range: [0, 1],
+          location_type: 'start',
+          range_type: 'distance',
+          units: 'mi'
+        })
+      })
+        console.log('Status:', response.status);
+        const data =  await response.json();
+        setDestinationRadius(data)
+      }
+      catch(error) {
+        console.error('Error:', error);
+      };
+    }
+
+
+
+
+
+
+    useEffect(() => {
+        fetchDirections();
+        fetchCurrentRadius()
+        fetchDestinationRadius()
+      }, []);
+
+
+      function midpoint(coord1, coord2) {
```

Comment:
Move this into a helper file? Also, I'd get rid of the nested functions of radians() and degrees() and bring them to the same scope-level as midpoint(..) in the helper function